[Kaptain]: NGC support (#4573)
mesosphere/dcos-docs-site: pages/dkp/kaptain/2.1.0/tutorials/ngc/index.md

---
layout: layout.pug
navigationTitle: NGC catalog
title: NVIDIA NGC catalog
menuWeight: 30
excerpt: Use NVIDIA's NGC catalog with Kaptain
beta: false
enterprise: false
---

Kaptain supports the usage of Nvidia's NGC catalog in networked environments, which you can access and set up via the Kubeflow UI. The NGC catalog includes containers, charts, pre-trained models, toolkits and more, which are optimized for GPU deployment and can now be used to run AI workflows in Kaptain. You can find all the available resources for your AI/ML workflow in the [NVIDIA NGC catalog][NGC_catalog].

> 

Author: Add a note here in the lines of: 

Not all NGC available images have been tested, so support is limited. 
install SDK or other (see slides from presi) for added funtionality.

## Prerequisites

-   You have [deployed Kaptain into a cluster](../../install/).

-   You have a Kubeflow profile, and [permissions to create notebooks](../../user-management/).

-   You have permissions to access the [NGC component](https://docs.nvidia.com/ngc/ngc-catalog-user-guide/index.html#registering-activating-ngc-account).

## Prepare your Kubeflow user profile

You do not need to download, nor install anything to be able to access the NGC catalog. You can visit NVIDIA's [NGC catalog page][NGC_catalog] directly or access it from your Kaptain instance. However, you have to set the default entry point for Jupyter Lab first.

In your Kubeflow profile, create a custom configuration for a notebook server to launch in Jupyter Lab. Specify the `user-namespace` where you want to create the notebook.

1.  Create or edit a `PodDefault` YAML file:

>     ```yaml
>     apiVersion: "kubeflow.org/v1alpha1"
>     kind: PodDefault
>     metadata:
>       name: custom-entrypoint
>       namespace: <kubeflow-profile>
>     spec:
>       selector:
>         matchLabels:
>           custom-entrypoint: "true"
>     desc: "Launch as Jupyter Lab"
>     command:
>     - jupyter
>     args:
>     - lab
>     - --notebook-dir=/workspace
>     - --ip=0.0.0.0
>     - --no-browser
>     - --allow-root
>     - --port=8888
>     - --NotebookApp.token=''
>     - --NotebookApp.password=''
>     - --NotebookApp.allow_origin='*'
>     - --NotebookApp.base_url=$(NB_PREFIX)
>     ```

Reviewer: ```suggestion
    ```yaml
    apiVersion: "kubeflow.org/v1alpha1"
    kind: PodDefault
    metadata:
      name: custom-entrypoint
      namespace: <kubeflow-profile>
    spec:
      selector:
        matchLabels:
          custom-entrypoint: "true"
      desc: "Launch as Jupyter Lab"
      command:
      - jupyter
      args:
      - lab
      - --notebook-dir=/workspace
      - --ip=0.0.0.0
      - --no-browser
      - --allow-root
      - --port=8888
      - --NotebookApp.token=''
      - --NotebookApp.password=''
      - --NotebookApp.allow_origin='*'
      - --NotebookApp.base_url=$(NB_PREFIX)
    ```
```

---
layout: layout.pug
navigationTitle: NGC catalog
title: NVIDIA NGC catalog
menuWeight: 30
excerpt: Use NVIDIA's NGC catalog with Kaptain
beta: false
enterprise: false
---

> Kaptain supports the usage of Nvidia's NGC catalog in networked environments, which you can access and set up via the Kubeflow UI. The NGC catalog includes containers, charts, pre-trained models, toolkits and more, which are optimized for GPU deployment and can now be used to run AI workflows in Kaptain.

Reviewer 2: We need to be careful with the word 'supports' - some readers might interpret it to mean that D2iQ will provide you with engineering support for all the stuff in the NGC, which I don't believe is our plan or intention.

```suggestion
Kaptain is compatible with the usage of Nvidia’s NGC catalog in networked environments, which you can access and set up via the Kubeflow UI; however, usage of NGC catalog components is not covered under D2iQ's support policy.   

The NGC catalog includes containers, charts, pre-trained models, toolkits and more, which are optimized for GPU deployment and can now be used to run AI workflows in Kaptain.
```

> Since the NGC catalog is a contantly growing third-party catalog developed by NVIDIA, not all available images have been tested. Note that for some examples, you need to install additional tools like TAO Toolkit, DeepStream SDK, NVIDIA TensorRT and others.

Reviewer 3: ```suggestion
Since the NGC catalog is a constantly growing third-party catalog developed by NVIDIA, not all available images have been tested. Note that for some examples, you need to install additional tools like TAO Toolkit, DeepStream SDK, NVIDIA TensorRT and others.
```

> Since the NGC catalog is a constantly growing third-party catalog developed by NVIDIA, not all available images have been tested. Note that for some examples, you need to install additional tools like TAO Toolkit, DeepStream SDK, NVIDIA TensorRT and others.

Reviewer 2: ```suggestion
Since the NGC catalog is a constantly growing third-party catalog developed by NVIDIA, not all available images have been tested.   Also, note that for some examples, you may need to install additional tools like TAO Toolkit, DeepStream SDK, NVIDIA TensorRT and others, which is beyond the scope of this documentation.  
```

> You can find all the available resources for your AI/ML workflow in the [NVIDIA NGC catalog][NGC_catalog], and detailed intructions on how to use these resources in the [NGC documentation][NGC_docs] site.

Author: ```suggestion
You can find all the available resources for your AI/ML workflow in the [NVIDIA NGC catalog][NGC_catalog], and detailed instructions on how to use these resources in the [NGC documentation][NGC_docs] site.
```

> You can find all the available resources for your AI/ML workflow in the [NVIDIA NGC catalog][NGC_catalog], and detailed instructions on how to use these resources in the [NGC documentation][NGC_docs] site.

Reviewer 2: I wonder about the word 'all' here, my first reading suggested that it meant I could find everything I would ever need in this catalog, which I doubt is true.    :-) 

```suggestion
You can find resources for your AI/ML workflow in the [NVIDIA NGC catalog][NGC_catalog], and detailed instructions on how to use these resources in the [NGC documentation][NGC_docs] site.
```

---
layout: layout.pug
navigationTitle: NGC catalog
title: NVIDIA NGC catalog
menuWeight: 30
excerpt: Use NVIDIA's NGC catalog with Kaptain
beta: false
enterprise: false
---

Kaptain supports the usage of Nvidia's NGC catalog in networked environments, which you can access and set up via the Kubeflow UI. The NGC catalog includes containers, charts, pre-trained models, toolkits and more, which are optimized for GPU deployment and can now be used to run AI workflows in Kaptain. You can find all the available resources for your AI/ML workflow in the [NVIDIA NGC catalog][NGC_catalog].

## Prerequisites

-   You have [deployed Kaptain into a cluster](../../install/).

-   You have a Kubeflow profile, and [permissions to create notebooks](../../user-management/).

-   You have [installed JupyterLab](https://jupyterlab.readthedocs.io/en/2.2.x/getting_started/installation.html).

-   You have permissions to access the [NGC component](https://docs.nvidia.com/ngc/ngc-catalog-user-guide/index.html#registering-activating-ngc-account).

## Prepare your Kubeflow user profile

> You do not need to download, nor install the NGC catalog. You will be able to access it by logging into your Kaptain instance. However, you have to set the default entry point for Jupyter Lab first.

Reviewer: > You will be able to access it by logging into your Kaptain instance

The catalog can be accessed by going to the https://catalog.ngc.nvidia.com/ page directly. You don't have to log in into Kaptain to access the catalog

> In your Kubeflow profile, create a custom configuration for a notebook server to launch in Jupyter Lab. Specify the `user-namespace` where you want to create the notebook.

Author: @alembiewski I saw in the demo that you need to carry out the following steps from the kudo-kubeflow profile, do customers know how to switch to that, or should we provide the command here?

1.  Open the `PodDefault` YAML file:

>     <!-- TODO: command to "open" or "edit" the yaml -->

Author: @alembiewski is there a command to open the yaml for editing? Could you provide it here, please? 
or should we go for something similar to `kubectl apply` where you create and apply a resource in one step? 
It seems to me as if the `PodDefault` resource is created for this purpose specifically?

Reviewer 2: They can use an text editor, there's no special command and I don't think we need to provide more detail here.

However, I think to keep it consistent with other places in our docs, this formulation might be more better:

Create or edit a 'PodDefault' YAML file:

    ```yaml
    apiVersion: "kubeflow.org/v1alpha1"
    kind: PodDefault
    metadata:
    name: custom-entrypoint
    namespace: <user-namespace>
    spec:
    selector:
        matchLabels:
        custom-entrypoint: "true"
    desc: "Launch as Jupyter Lab"
    command:
    - jupyter
    args:
    - lab
    - --notebook-dir=/workspace
    - --ip=0.0.0.0
    - --no-browser
    - --allow-root
    - --port=8888
    - --NotebookApp.token=''
    - --NotebookApp.password=''
    - --NotebookApp.allow_origin='*'
    - --NotebookApp.base_url=$(NB_PREFIX)

>     ```

Reviewer 2: This doesn't look to be formatted correctly, in particular, the indenting in the selector spec looks wrong to me.   (I believe custom-entrypoint should be indented at least 2 spcs more...)

1.  Run this command to apply the new entry point:

    ```bash

>     k apply -n user -f pd.yaml

Reviewer 2: They wouldn't need to specify the namespace (-n ) as it is in the .yaml file.    Also, we should not use aliases or shortcuts in commands like this.  'k' is an alias that many developers use to avoid typing 'kubectl', but for completeness we should spell it out.

```suggestion
    kubectl apply -f podddefault.yaml
```
